pkg/clusterapi/registry_mirror.go

```diff
@@ -54,18 +55,19 @@ func SetRegistryMirrorInKubeadmConfigTemplateForUbuntu(kct *bootstrapv1.KubeadmC
 
 func registryMirror(mirrorConfig *v1alpha1.RegistryMirrorConfiguration) bootstrapv1.RegistryMirrorConfiguration {
 	return bootstrapv1.RegistryMirrorConfiguration{
-		Endpoint: net.JoinHostPort(mirrorConfig.Endpoint, mirrorConfig.Port),
+		Endpoint: containerd.ToAPIEndpoint(mirrorConfig.RegistryMirror().RegistryMirrorWithOCINamespace()),
 		CACert:   mirrorConfig.CACertContent,
 	}
 }
 
 type values map[string]interface{}
 
-func registryMirrorConfigContent(registryAddress, registryCert string, insecureSkip bool) (string, error) {
+func registryMirrorConfigContent(registryMirror *registrymirror.RegistryMirror, registryCert string, insecureSkip bool) (string, error) {
 	val := values{
-		"registryMirrorAddress": registryAddress,
-		"registryCACert":        registryCert,
-		"insecureSkip":          insecureSkip,
+		"registryMirrorMap": containerd.ToAPIEndpoints(registryMirror.NamespacedRegistryMap),
+		"mirrorBase":        registryMirror.BaseRegistry,
+		"registryCACert":    registryCert,
```

Comment:
I have the same thoughts. We can gradually move all registry mirror related stuff to this struct, e.g., username and password. Since this PR is already big enough, those improvements can be made in separate smaller PRs in the future.

Reply:
Sounds good. Could you create a ticket to track this follow up items, please?

- [ ] Add registry mirror cert to `registrymirror.RegistryMirror` and possibly credentials as well